specification/security/resource-manager/Microsoft.Security/stable/2024-08-01/customRecommedations.json

```diff
@@ -0,0 +1,473 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Microsoft Defender for Cloud",
+    "description": "API spec for Microsoft.Security (Microsoft Defender for Cloud) resource provider",
+    "version": "2024-08-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/{scope}/providers/Microsoft.Security/customRecommendations": {
```

Comment:
>/{scope} [](http://example.com/codeflow?start=5&length=8)

Consider enumerating supported scopes. it's better to identify the scopes individually in the spec, rather than club them all together under a single parameter.

Reply:
We are using the same pattern in all of our microsoft.security APIs, this is the way our customers used to.
Also, the use case here is similar to microsoft.autorization policy APIs which used the same pattern.